### DISCUSSION TITLE
Feature proposals concerning quiz experience.

### DISCUSSION THREAD

**Discussion by drvnprgrmr** [freeCodeCamp/Developer_Quiz_Site#837] [Ideas]
I have two suggestions (might have more) on some ways to improve the quiz experience.

1. Don't submit questions immediately they're clicked on just in case the user made a mistake or had a change of heart at the last moment.
This would probably require adding something like a submit button and managing state of currently selected questions.

2. Allow a user to "finish" prematurely. 
e.g. In a scenario where a user selects a round of 100 questions and something comes up midway that won't allow the user to complete their round, the user should be able to quit the current session and view their scores.

**Comment by drvnprgrmr** [comment]
Actually just realized that there is a score counter so I guess **2.** isn't needed after all.

**Comment by jdwilkin4** [comment]
So I think it would work to essentially still record their responses but then create a back button.
So if a user wants to go back a question, then they can with the back button and update their answer. Then we just update that answer for that particular question.

**Reply by drvnprgrmr** [reply]
Not really sure I understand but do you mean that a user can change their answer *after* seeing the correct one?

**Reply by jdwilkin4** [reply]
> do you mean that a user can change their answer after seeing the correct one?

I forgot about the pop up modal.

In this case, it sounds like your solution would mean getting rid of the pop up modal if we are planning on changing how the questions are submitting at the end.
Which, I am not a fan of removing the modal.
I like the current behavior of showing the answer with the explanation after each question.

If you can find another solution, that means we can keep the modal after each question then I am all ears 👍

**Reply by drvnprgrmr** [reply]
I didn't intend on removing the modal at all. Just delaying it.

**Reply by drvnprgrmr** [reply]
Instead of showing it immediately we temporarily highlight the option picked and then when the user clicks on another button we continue with the flow.

**Reply by jdwilkin4** [reply]
So then would you esstentially have one giant modal at the end of the quiz displaying all of the correct answers with explanations?

**Reply by drvnprgrmr** [reply]
Not at the end of the whole quiz. Just at the end of a question. 
It's just a slight change from the current flow.

**Reply by jdwilkin4** [reply]
Alright, I understand what you are after now.

You can go ahead and create a detailed issue in the issues list and then I can assign it to you.

**Reply by drvnprgrmr** [reply]
Okay thanks.
